applylib/go.mod

```diff
@@ -14,6 +14,7 @@ require (
 	k8s.io/klog/v2 v2.90.1
 	sigs.k8s.io/controller-runtime v0.14.1
 	sigs.k8s.io/kubebuilder-declarative-pattern/mockkubeapiserver v0.0.0-20221021151406-9bd3fb842119
+	sigs.k8s.io/kustomize/kstatus v0.0.2-0.20200509233124-065f70705d4d
```

Comment:
kstatus is under cli-utils (originally in kustomize) and we do not want to import cli-utils in applyset. This workaround  is to use the kstatus when it was a standalone go library in kustomize. This introduces the minimum dependencies.

Related libraries:
kstatus is removed from kustomize in api/v0.4.2 https://github.com/kubernetes-sigs/kustomize/pull/2674
this PR uses the commit from api/v0.4.1

Reply:
Seems reasonable!